### DISCUSSION TITLE
ToolTips on fields should start with "Specifies "

### DISCUSSION THREAD

**Discussion by jwikman** [StefanMaron/BusinessCentral.LinterCop#346] [New Rule]
According to the guidelines for ToolTips (https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/user-assistance#guidelines-for-tooltip-text), all ToolTips on field controls should start with "Specifies".

The rule should check that the ToolTip, if specified, for fields on pages (and request pages on reports, I assume) starts with "Specifies ".

**Comment by Arthurvdv** [comment]
Added this to the existing LC0026 rule and is now available in the (pre)release version [v0.30.8](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.8).

**Reply by jwikman** [reply]
Thanks @Arthurvdv!
And a very wise move to add it to the existing rule. 👍

**Reply by jwikman** [reply]
@Arthurvdv just tested this out, but it seems as if LC0026 got broken with this update (v0.30.8)


This error is thrown about 1000+ times, together with approx the same numbers of false positives of LC0026. (also a lot of false positives on LC0028, don't know if that is related though)
```
[{
	"resource": "/d:/VSCode/Git/NAB/Eagle/App/app.json",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": "AD0001",
	"severity": 4,
	"message": "Analyzer 'BusinessCentral.LinterCop.Design.Rule0026ToolTipPunctuation' threw an exception of type 'System.InvalidCastException' with message 'System.InvalidCastException: Unable to cast object of type 'Microsoft.Dynamics.Nav.CodeAnalysis.Symbols.SourceActionSymbol' to type 'Microsoft.Dynamics.Nav.CodeAnalysis.IControlSymbol'.\r\n   at BusinessCentral.LinterCop.Design.Rule0026ToolTipPunctuation.AnalyzeToolTipPunctuation(SyntaxNodeAnalysisContext ctx)\r\n   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass52_1.<ExecuteSyntaxNodeAction>b__1() in D:\\a\\_work\\1\\s\\source\\Prod\\Microsoft.Dynamics.Nav.CodeAnalysis\\DiagnosticAnalyzer\\AnalyzerExecutor.cs:line 745\r\n   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info) in D:\\a\\_work\\1\\s\\source\\Prod\\Microsoft.Dynamics.Nav.CodeAnalysis\\DiagnosticAnalyzer\\AnalyzerExecutor.cs:line 1102'",
	"source": "AL",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 1
}]

```

**Reply by Arthurvdv** [reply]
Thanks for reporting! Should be fixed now.

**Reply by jwikman** [reply]
Great, thanks!

And a v.0.30.9 is baking in the oven? 🙂

**Reply by Arthurvdv** [reply]
No problem!

Prereleases don’t get a new version number on updates. So it’s still the same version, but without the error hopefully 🤞

**Reply by jwikman** [reply]
Aaah, I see! 👍

This issue is now gone, but LC0028 is giving all false positives. Should I create a new issue for that one? Don't have much more repro info than "it does not work." 😅 (the warning shows up, even when the string literal been changed to an identifier)

**Reply by Arthurvdv** [reply]
Yeah, best a new issue for the LC0028 rule you’re experiencing problems with. I’ll have a look into that tomorrow.
